lib/fathead/flask/parse.py

```diff
@@ -0,0 +1,182 @@
+        return self.FILE
+
+
+class Parser(object):
+    """
+    Parser to parse HTML files
+    """
+    
+    def __init__(self, data_object):
+        self.file_being_used = data_object.get_file()
+        self.file_name = self.file_being_used.split('/').pop()
+        self.sections = []
+        self.soup = BeautifulSoup(data_object.get_raw_data(), 'html.parser')
+
+    def parse_for_data(self):
+        for header in self.soup.find_all(['dl']):
+            section = self.parse_section(header)
+            section['example'] = self.clean_formatting(self.get_example(header))
+            if section['paragraph'] and section['example']:
+                section['abstract'] = '<section class="prog__container><p>{}</p><pre><code>{}</code></pre></section>'.format(section['paragraph'], section['example'])
+                self.sections.append(section)
+            elif section['paragraph'] and not section['example']:
+                section['abstract'] = '<section class="prog__container><p>{}</p></section>'.format(section['paragraph'])
+                self.sections.append(section)
+
+        self.parsed_data = self.sections
+
+    def get_data(self):
+        return self.parsed_data
+
+    def get_example(self, soup):
+        text_html = soup.findNext('pre')
+        if not text_html is None:
+            text = text_html.get_text()
+            text = text.strip('\n')
+            return text.replace('\n', '\\n')
+        else:
+            return ""
+
+    def get_url(self, soup):
+
+        link_html = soup.find('a', href=True)
+        if not link_html is None:
+            anchor=link_html["href"]
+        else:
+            anchor=''
+
+        if anchor:
+            parsed_url = '{}{}'.format(os.path.join(URL_ROOT, self.file_name), anchor)
+        else:
+            parsed_url = os.path.join(URL_ROOT, self.file_name)
+        
+        return parsed_url
+
+    def parse_section(self, soup):
+        first_paragraph_html = soup.findNext('p')
+        if not first_paragraph_html is None:
+            first_paragraph = self.clean_formatting(first_paragraph_html.get_text())
+        else:
+            first_paragraph=''
+        title=''
+        for el in soup.find_all("dt", id=True):
+            title = el.get('id')
+        if title == '':
+            for el in soup.find_all("code"):
+                title+=el.get_text()
+        anchor = self.get_url(soup)
+        url = os.path.join(URL_ROOT, self.file_name)
+
+        return {
+            'title': title,
+            'paragraph': first_paragraph,
+            'anchor': anchor,
+            'url': url
+        }
+
+    def clean_formatting(self, text):
+        
+        """
+        Flask documentation contains '¶' symbol to represent hyperlinks
+        which needs to be removed. Double spaces needs to be 
+        converted to single spaces 
+        """
+        
+        text = text.replace('  ', ' ').replace('\n', ' ').replace('\\n', r'\\n').replace("Â¶","").strip()
```

Comment:
You probably want to remove any `\n` inside `<p>..</p>` after as well. That way we'll let the text wrap inside paragraphs, but maintain the newlines as they appear in the code 👍